src/bridge/EspressoTEEVerifier.sol

```diff
@@ -14,50 +14,85 @@ import {
+    using BytesUtils for bytes;
 
-contract EspressoTEEVerifier is V3QuoteVerifier {
-    constructor(address _router) V3QuoteVerifier(_router) {}
+    // V3QuoteVerififer contract from automata to verify the quote
+    V3QuoteVerifier public quoteVerifier;
+    bytes32 public mrEnclave;
+    bytes32 public mrSigner;
 
-    /**
+    constructor(bytes32 _mrEnclave, bytes32 _mrSigner, address _quoteVerifier) {
+        quoteVerifier = V3QuoteVerifier(_quoteVerifier);
+        mrEnclave = _mrEnclave;
+        mrSigner = _mrSigner;
+        _initializeOwner(msg.sender);
+    }
+
+    /*
         @notice Verify a quote from the TEE and attest on-chain
         @param rawQuote The quote from the TEE
-        @return success True if the quote was verified and attested on-chain
-     */
-    function verify(bytes calldata rawQuote) external view returns (bool success) {
+        @param reportDataHash The hash of the report data
+        @return success True if the quote was verified and attested on-chain, false otherwise
+    */
+    function verify(
+        bytes calldata rawQuote,
+        bytes32 reportDataHash
+    ) external view returns (bool success) {
         // Parse the header
-        Header memory header = _parseQuoteHeader(rawQuote);
+        Header memory header = parseQuoteHeader(rawQuote);
 
+        // Currently only version 3 is supported
         if (header.version != 3) {
             return false;
         }
 
-        (success, ) = this.verifyQuote(header, rawQuote);
+        // Verify the quote
+        (success, ) = quoteVerifier.verifyQuote(header, rawQuote);
         if (!success) {
             return false;
         }
 
-        //  Parse enclave quote
+        // // Parse enclave quote
         uint256 offset = HEADER_LENGTH + ENCLAVE_REPORT_LENGTH;
         EnclaveReport memory localReport;
         (success, localReport) = parseEnclaveReport(rawQuote[HEADER_LENGTH:offset]);
         if (!success) {
             return false;
         }
 
-        return true;
+        // Check that mrEnclave and mrSigner match
+        if (localReport.mrEnclave != mrEnclave || localReport.mrSigner != mrSigner) {
+            return false;
+        }
+
+        //  Verify that the reportDataHash if the hash signed by the TEE
+        // We do not check the signature because `quoteVerifier.verifyQuote` already does that
+        if (reportDataHash != bytes32(localReport.reportData.substring(0, 32))) {
+            return false;
+        }
 
-        // TODO: Use the parsed enclave report (localReport) to do other verifications
+        return true;
     }
 
-    function _parseQuoteHeader(
-        bytes calldata rawQuote
-    ) private pure returns (Header memory header) {
+    /*
+        @notice Parses the header from the quote
+        @param rawQuote The raw quote in bytes
+        @return header The parsed header
+    */
+    function parseQuoteHeader(bytes calldata rawQuote) public pure returns (Header memory header) {
```

Comment:
I didnt see any harm in making it public and I think it might turn out helpful for debugging in the future.  Its a pure function so takes very less gas as well.